### PR TITLE
Refactor: use bulkWrite in updateJudge for improved efficiency

### PR DESCRIPTION
**Pull request by samrat-rm**
### Description

refactor(updateJudge): replace individual updates with bulkWrite operation
- Switched to bulkWrite in updateJudge function to handle updates more efficiently.
- Reduces the number of database calls and improves performance.

### Type of Change 

Delete options that do not apply:

- Refactor (code changes that doesn't affect functionality)

### Is this a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by samrat-rm** [comment]
@MichaelZhao21  Please review this PR for issue #192

**Comment by MichaelZhao21** [comment]
@samrat-rm commenting on a PR is not going to merge your PR faster.

**Comment by samrat-rm** [comment]
> @samrat-rm commenting on a PR is not going to merge your PR faster.

Sorry, This is my first OS contribution and I am not sure what to do for my PR to get reviewed and merged.

File: server/database/judge.go [hunk 1]
```diff
@@ -87,13 +88,13 @@ func UpdateJudges(db *mongo.Database, sc, judges []*models.Judge) error {
...
 	for _, judge := range judges {
-		err := UpdateJudge(db, judge, sc)
-		if err != nil {
-			return err
-		}
+		models = append(models, mongo.NewUpdateOneModel().SetFilter(gin.H{"_id": judge.Id}).SetUpdate(gin.H{"$set": judge}))
 	}
-	return nil
+	opts := options.BulkWrite().SetOrdered(false)
+	_, err := db.Collection("judges").BulkWrite(context.Background(), models, opts)
```
Comment (MichaelZhao21) [inline server/database/judge.go:96]: Instead of using `context.Background`, you should use the provided `sc` mongo session context.

**Review by MichaelZhao21 — CHANGES_REQUESTED** [review]

**Comment by MichaelZhao21** [comment]
@samrat-rm what is the status on the changes?

**Comment by samrat-rm** [comment]
> @samrat-rm what is the status on the changes?

completed 👍  
Please review

**Review by MichaelZhao21 — APPROVED** [review]

**Pull request merged**